include/CL/sycl/atomic.hpp

```diff
@@ -68,11 +68,33 @@ class atomic {
   /// \todo unimplemented
   HIPSYCL_KERNEL_TARGET
   void store(T operand, memory_order memoryOrder =
-      memory_order::relaxed) volatile;
+      memory_order::relaxed) volatile
+  {
+#ifdef __HIPSYCL_DEVICE_CALLABLE__
+#ifdef SYCL_DEVICE_ONLY
+    return detail::invalid_host_call_dummy_return<T>();     // really should be invalid_device_call
```

Comment:
Better not call this on device. This function calls `assert()` which may not work on all GPUs (I think newer NVIDIA GPUs/newer CUDA versions may be able to do that, not sure about AMD).

You could instead surround the entire function with `#ifdef HIPSYCL_PLATFORM_CPU` to make it only visible when compiling for CPU.

Reply:
That would require moving it outside the class declaration, right?

Reply:
Would a `static_assert` be better?

Reply:
Actually, never mind, it's not a template function, `static_assert` would get triggered every time.

Reply:
I think an out-of-line definition outside of the class would make sense. In this case the declaration could be visible both on CPU and GPU, but the definition would be invisible on GPU, leading to a linker error.

Reply:
How about now?

Reply:
looks good, thank you!